dockerfiles/cli/cli_tests.bats

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env bats
+# Copyright (c) 2012-2017 Red Hat, Inc
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# HOW TO RUN TEST?
+# To run tests, ensure you have:
+#     - installed bats
+#     - running docker (best experience v 1.12 and higher)
+#     - optional - clean docker ('docker ps' and 'docker ps -a' are empty)
+#         
+# Then run command 'bats cli_tests.bats' 
+# from the directory containing these bats tests.
+#
+# Contributors:
+#   Marian Labuda - Initial Implementation
+#
+#
+
+source ../base/scripts/base/startup_funcs.sh
+export ALL_TEST_PATH=$(pwd)/testrun
+export TEST_DIR_PATH=$ALL_TEST_PATH/testdata
+mkdir $ALL_TEST_PATH -p
+
+# set up environment
+setup() {
+  clean_docker
+} 
+
+clean_docker() {
+  if [ "$(docker ps -q | wc -l)" -gt 0 ]; then
+    docker kill $(docker ps -q)
+  fi
+  if [ "$(docker ps -qa | wc -l)" -gt 0 ]; then
+    docker rm $(docker ps -qa)
```

Comment:
we could have requirements performed in a test as well.
Like if port 8080 is not available then it fails the requirements
or find a random port then start che on this random avilable port instead of using 8080